app/scripts/lib/logger.js

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Shared logging module that performs the checks and string
+ * interpolation.
+ *
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  /**
+   * Constructor of log module.
+   *
+   * @param logWindow Window object that contains console used for logging.
+   *
+   * @constructor
+   */
+  function Logger(logWindow) {
+    this._window = logWindow || window;
+  }
+
+  Logger.prototype = {
+
+    /**
+     * Wrapper for `console.info` function that checks for availability and
+     * then prints info.
+     *
+     */
+    info: function () {
+      if (this._window.console && this._window.console.info) {
```

Comment:
Or if the concern is that the `console` object will be modified at some later point, what if we refactor the checks and the `apply` call to one place and pass in the method name as an argument?

Something like:

```js
  function createLoggerMethod (name) {
    return function () {
      if (this._window.console && this._window.console[name]) {
        this._window.console[name].apply(this._window.console, arguments);
      }
    }
  }

  // ...

  info: createLoggerMethod('info'),
  warn: createLoggerMethod('warn'),

  // etc
```

Or maybe I'm overthinking it, feel free to ignore. :grin: